### DISCUSSION TITLE
Implementing Lasso Select - Looking for tips

### DISCUSSION THREAD

**Discussion by MxMossy** [excalidraw/excalidraw#6494] [General]
Hello, I was interested in taking a stab at implementing the lasso selection described [here](https://github.com/excalidraw/excalidraw/issues/6350). I've been going through the code base to try and understand the current selection tool but was wondering if anyone had pointers for where I should start looking.

This was my current idea of implementation:
- render a polygon to represent the lasso selection area (was thinking to use alt key but seems alt mirrors the selection aread about the origin already).
- perform logic checks to see what objects are enclosed

Any other pointers would be appreciated, thanks!

**Comment by dwelle** [comment]
Hi. I've marked the issue as `discussion` so we first decide how it will work and how we would expose that feature.

**Reply by dwelle** [reply]
Agree on the alt-key, let's use it. The mirroring behavior is just an unintended side effect of using a pseudo element for selection.
